editor/store/effects.js

```diff
@@ -96,15 +97,15 @@ export default {
 	REQUEST_POST_UPDATE( action, store ) {
 		const { dispatch, getState } = store;
 		const state = getState();
-		const isAutosave = action.options && action.options.autosave;
+		const post = getCurrentPost( state );
+		const isAutosave = get( action.options, [ 'autosave' ], false );
 
 		// Prevent save if not saveable.
 		const isSaveable = isAutosave ? isEditedPostAutosaveable : isEditedPostSaveable;
 		if ( ! isSaveable( state ) ) {
 			return;
 		}
 
-		const post = getCurrentPost( state );
 		const edits = getPostEdits( state );
 		let toSend = {
 			...edits,
```

Comment:
We know that the server does not save all the fields during autosave should we pick here the fields we send to the server, in order not have unused bytes sent to the server?

Reply:
>We know that the server does not save all the fields during autosave

Do we know this? I sincerely don't know and would be curious to know the answer. In theory it seems it should only need to save these fields, but in the implementation it's just calling `wp_update_post` and doesn't have its own implementation of `prepare_item_for_database`, falling back instead to `WP_REST_Revisions_Controller::prepare_item_for_database`. It's hard to follow if these fields might be dropped by part of a schema definition in one of the autosaves, revisions, or post controller classes.

https://github.com/WordPress/gutenberg/blob/102445bf6ec65a58b5002fb07ad7b98b6c86655c/lib/class-wp-rest-autosaves-controller.php#L185

That said, to the [previous conversation](https://github.com/WordPress/gutenberg/pull/7130#discussion_r193546465), it might make things simpler if we could assume that at most, only title, excerpt, and content would be updated in order to know how to update `state.currentPost`.

Reply:
>Do we know this? I sincerely don't know and would be curious to know the answer.

I tested this, and for a new post, if I change author and hit preview (i.e. autosave), the author is updated.

Reply:
> We know that the server does not save all the fields during autosave

Nearly all fields (at least date, date_gmt, post_title, post_content, post_excerpt, post_modified, post_modified_gmt), are saved to the autosave as a new wp_posts row linked to the original post by a post_parent column.  some fields get a special value: post_name (usually the slug) is stored as [POST_ID]-autosave-v1, post_status is inherit, post_type is revision.

I'm not certain about to_ping, pinged, post_password, ping_status, comment_count, menu_order, comment_status - I think these are saved. guid is a unique id pointing to the autosave.

related - attached meta fields are _currently saved to the original post_ a longstanding core bug, see https://core.trac.wordpress.org/ticket/20299 - we recently fixed this for featured image selection, however a more complete implementation would need to support revisioning of meta, since we need to attach the saved meta to the autosave for previewing. see https://github.com/adamsilverstein/wp-post-meta-revisions and https://core.trac.wordpress.org/ticket/20564